### PR TITLE
Add optional Pipeline integration-in-Kind Prow job

### PR DESCRIPTION
**Pull request by abayer**
# Changes

This will need https://github.com/tektoncd/pipeline/pull/5077 to be merged before it will actually work with any other PRs, but that's fine. =)

Assuming this works, I'll be adding the alpha integration job as well in a follow up.

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: prow/config.yaml [hunk 1]
```diff
@@ -1211,6 +1211,51 @@ presubmits:
...
+  - name: pull-tekton-pipeline-kind-integration-tests
+    labels:
+      preset-presubmit-sh: "true"
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    agent: kubernetes
+    always_run: false
+    decorate: true
+    optional: true
+    rerun_command: "/test pull-tekton-pipeline-integration-tests"
```
Comment (afrittoli) [inline prow/config.yaml:1223]: should this be aligned to the trigger or is it on purpose?

Comment (abayer) [reply prow/config.yaml:1223]: That is a mistake. Fixing.

**Pull request merged**